### PR TITLE
Warn for duplicate scripts loaded

### PR DESCRIPTION
**Pull request by bassettsj**
Great package, simple and works well. Thanks for writing it. 

I wanted to purpose adding a small check to make sure the script hasn't already been loaded once before.

- Added a utility function to check `containsScript(src)`
- Added unit test for that condition

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -63,3 +68,18 @@ function ieOnEnd (script, cb) {
...
+
+
+function containsScript(src) {
+  var script;
+  var scripts = document.scripts;
+  for (var i = 0; i <= scripts.length; i++) {
+    script = scripts[i];
+
+    if (script && script.src && script.src == src) {
+
```
Comment (defunctzombie) [inline index.js]: extra line

File: index.js [hunk 2]
```diff
@@ -11,6 +11,11 @@ module.exports = function load (src, opts, cb) {
   opts = opts || {}
   cb = cb || function() {}
 
+  // Throw error if Document already has script loaded.
+  if (containsScript(src, document)) {
```
Comment (defunctzombie) [inline index.js]: document var not used in function

Comment (bassettsj) [inline index.js]: Thank you for pointing out, fixed.

File: index.js (hunk 1 above)
Comment (bassettsj) [inline index.js]: Fixed.

**Comment by defunctzombie** [comment]
In thinking more about this, what if you want to load the same script file again? Maybe because it changed?

**Comment by bassettsj** [comment]
True. It could be possible to want to do that. So let's just close this.

**Comment by defunctzombie** [comment]
thanks for experimenting tho!

If you get the urge to experiment more:
https://github.com/eldargab/load-script/issues/11 :smile:

**Pull request closed without merging**